packages/headless/src/features/generated-answer/generated-answer-slice.ts

```diff
@@ -68,12 +76,22 @@ export const generatedAnswerReducer = createReducer(
         state.feedbackSubmitted = true;
       })
       .addCase(resetAnswer, (state) => {
-        return {
-          ...getGeneratedAnswerInitialState(),
-          responseFormat: state.responseFormat,
-          fieldsToIncludeInCitations: state.fieldsToIncludeInCitations,
-          isVisible: state.isVisible,
-        };
+        return handleResetAnswer(state);
+      })
+      .addCase(executeSearch.fulfilled, (state, payload) => {
+        const streamId =
+          payload.payload.response.extendedResults
+            .generativeQuestionAnsweringId;
+        return state.currentStreamId === streamId
+          ? state
+          : {
+              ...handleResetAnswer(state),
+              shouldStartStream: !!streamId,
+              currentStreamId: streamId,
```

Comment:
When the stream ID is truthy, I think we should set `shouldAbortStream` to false.
Since by setting `shouldStartStream ` to true the ASM listener will already abort the previous stream, no?
```suggestion
              ...handleResetAnswer(state),
              shouldStartStream: !!streamId,
              shouldAbortStream: !streamId,
              currentStreamId: streamId,
```